### ISSUE TITLE
lynx knowledge navigator

### ISSUE THREAD

**Issue by Adapts** [Jermolene/TiddlyWiki5#1733]
'pologies in advance if this is only my issue.

ok, i have thousands of .tid files in a tree representing bookmarks.   i'd like to use '5' to generate densly linked pages usable with lynx -- lynx controlling what i see in ffox.  these should update as i dump more .tids into the tree from the outside - and when new .tids are created by node through the interface.

its not clear to me how to realize this. how can i reach enlightenment?

**Comment by Jermolene** [comment]
Hi @Adapts I'm not sure what you are referring to by "Lynx"; do you mean the text-based browser?

Anyhow, if you've got data that you want to appear in a tree you should look at the core table of contents macro:

http://tiddlywiki.com/#Table-of-Contents%20Macros

**Comment by Adapts** [comment]
Hi @Jermolene, thanks for jumping in. 

and yes, i use lynx. pretty much because browsers as i know them make for bad knowlede trackers.

tiddlywiki comes in because .tids can perfectly encapsulate knowledge .. and with the jump you've made to a no-database client-server architecture i want to port my link-pages into your wiki.

so far I use a mime-type handler to control lynx and another to populate my bookmark tracker from ffox.

lynx can exec shell links directly so that ffox gets told what to show. this way i can use a full screen browser on a side monitor controlled from a split terminal via keyboard and lynx.  this lets me use manual dense text pages to interact with the web instead of using browser-bookmarks.

its like having the web in an iframe or frameset, just different.

ok.. so now i want to get a M:1:M navigator going that works on dynamic javascript-free pages - without driving --rendertiddlers from the outside for every change in the generated .tids collection. 

**tl:dr;** is there a design document / api -reference that can help me understand how you're thinking about the capabilities of the current server .. does it have a full knowledge representation in memory which can  --rendertiddler a single thing statically on demand?

**Comment by Jermolene** [comment]
Hi @Adapts 

Just to be clear, am I right in understanding that you'd like to maintain your links as a bunch of static .tid files, and be able to run a script that generates a separate HTML file for each link tiddler, with links in between the HTML files?

>  is there a design document / api -reference that can help me understand how you're thinking about the capabilities of the current server .. does it have a full knowledge representation in memory which can --rendertiddler a single thing statically on demand?

TW works by loading tiddlers into memory at startup so that they are available to subsequent renderings. If you're producing static renderings, the usual approach would be to load up the wiki with the required tiddlers, and then use rendertiddler(s) to generate the required static HTML files. You should also be able to use the `toc` macro to create an interactive outline of your links.

**Comment by Adapts** [comment]
@Jermolene yes.

its 20 years later and i'm still inspired by **Jorn Barger's robotwisdom.com**.  example here:
[https://web.archive.org/web/20130409045039/http://www.robotwisdom.com/jaj/portal.html](https://web.archive.org/web/20130409045039/http://www.robotwisdom.com/jaj/portal.html)

i have ~7700 links/hosts/topics/tags &snippets curated around ~80 topic pages like the above. some of the content is added by hand @ around 5 links a day. some is auto-generated from places.sqlite, .history, /usr/share/doc, org-mode & dpkg.

right now it splices .tids into a tiddlywiki template and uses git-hooks/diffs to poke changes back into the fs. _its all rendered into the file-system_. often using empty files with extensions to indicate actions. this gives me density and speed, but not looks and hints.

my project refuses to go db/rest/xhtmlrequest exclusively as i want it support ext2/file-system/html2/text-browsing on ancient devices going forward also.

**tl;dr:** thanks for giving me a piece of your mind - and putting the server into place!  i hope it was fun to see how my work wants to bridge back to the static world. i might yet have a 404_to_wayback, lynxExec, and render2ext plugin for you before the decade is out.

/close

**Comment by Jermolene** [comment]
> right now it splices .tids into a tiddlywiki template and uses git-hooks/diffs to poke changes back into the fs. its all rendered into the file-system. often using empty files with extensions to indicate actions. this gives me density and speed, but not looks and hints.

Very interesting. I like the ability to stitch TiddlyWiki HTML files together using standard text tools; it takes the web back to the Unix command line.

**Comment by Adapts** [comment]
stitching is fine but the workflow of maintaining a tiddlywiki through a tiddlywiki has never been my thing. that's probably why i've lurked on your list for so long.  its all about org mode now. 

look at these options http://orgmode.org/manual/External-links.html#External-links .. i've used less than half of them but *99%* of my bookmarks land in org, with captured text that winds up curated and dumped back out to disk so that i can haz lynx execute and/or open things. see: http://lynx.isc.org/lynx2.8.6/lynx2-8-6/lynx_help/lynx_url_support.html#cgi_url (best feature ever)

org-mode export isn't my cup of tea though. that's why i write the tids and jump pages into the ln -s hyperlinked file-system with the goal of having tiddly make it pretty as can be .. both static - when used with mc (lol, seriously) or a no js html 2 browser .. and dynamic through tw5.

as software is becoming saner, wikis (for me) need to document the edge where fact & experience turn into learning/references.  cause there are too many simple pieces to keep in mind in the time between vacations. :)

**Comment by pmario** [comment]
@Jermolene ... label: "discussion"

**Comment by dubiouscript** [comment]
> Very interesting. I like the ability to stitch TiddlyWiki HTML files together using standard text tools; it takes the web back to the Unix command line.

was some method ever created or discovered to `stitch TiddlyWiki HTML files together using standard text tools` !??

... could `pup` and or `jq` 'stitch TiddlyWiki  files together' ?

+https://github.com/ericchiang/pup - Parsing HTML at the command line 
+https://github.com/stedolan/jq - Command-line JSON processor

**Comment by pmario** [comment]
@dubiouscript tiddlywiki.js is/has a CLI (command line interface). 
see: https://tiddlywiki.com/#Commands

Every TW edition can be built with eg: `tiddlywliki --build index` command. Every [edition](https://github.com/Jermolene/TiddlyWiki5/tree/master/editions/tw5.com) has a `tiddlywiki.info` file, which contains the "subcommands" that are used to build the eg: index file. 

The "tiddlers" directory contains all the content as text "dot-tid" files.

**Comment by dubiouscript** [comment]
@pmario thanks 
+ftr 
i found  empty.html easer to read / grok using  
```  cat empty.html  | ./pup  -p ```

**Comment by pmario** [comment]
@dubiouscript There would be a possibility to concatenate html files together. ... There is an ancient .tiddler format. `application/x-tiddler-html-div` which can still be used. ... But be aware, there is a difference between TWclassic and TW5!

see: https://tiddlywiki.com/#TiddlerFiles

**Comment by dubiouscript** [comment]
> @dubiouscript There would be a possibility to concatenate html files together. ... 
> see: https://tiddlywiki.com/#TiddlerFiles

im currently i am
 less concatenating html files together
more decomposing / filtering ./empty.html into fragments / fragmented views :smile: 

```
 cat ./empty.html  | \ 
pup -p '#storeArea :first-child pre text{}' | \
jq -cr '.tiddlers as $t | ($t|keys) as $k | .[]|.[]|keys  ' | \
tr -d '[]"' | tr ',' $'\n' | sort | uniq -c
 
      1 author
      5 button-classes
    173 caption
      4 color
     33 condition
      1 core-version
      5 created
      2 custom-icon
    141 description
     12 dropdown
      4 extension
     13 filter
      2 footer
     13 group
     13 group-sort
      2 help
     37 icon
      3 key
     19 list
      2 list-after
      3 list-before
      5 modified
    300 module-type
     23 name
     24 shortcuts
      3 subtitle
    353 tags
   1817 text
   1846 title
    315 type
      1 url
```

...and trying a little to better understand : TiddlyWiki5/core/modules/filters.js 
and related i/o
:loop:
in the hope being able to draw some parallels between the functionality of tiddly wiki filters and jq filters 

  :thinking:

**Comment by dubiouscript** [comment]
>  i can haz lynx execute and/or open things. see: http://lynx.isc.org/lynx2.8.6/lynx2-8-6/lynx_help/lynx_url_support.html#cgi_url (best feature ever)

https://lynx.invisible-island.net/lynx_help/lynx_url_support.html#cgi_url

:dark_sunglasses:
